packages/agent/src/oidc.ts

```diff
@@ -600,18 +594,116 @@ function encryptAuthResponse({
   return compactJwe;
 }
 
+export async function createPermissionGrants(
+  selectedDids: string[],
+  ephemeralDid: BearerDid,
+  dwn: AgentDwnApi
+) {
+  const grantPromises = selectedDids.map(async (did) => {
+    // TODO: Replace with real permission request
+    const permissionRequestData = {
+      description : 'The app is asking to Records Write to http://profile-protocol.xyz',
+      scope       : {
+        interface : DwnInterfaceName.Records,
+        method    : DwnMethodName.Write,
+        protocol  : 'http://profile-protocol.xyz',
+      },
+    };
+
+    // TODO: Confirm this
+    const message = await dwn.processRequest({
```

Comment:
@LiranCohen would know how it's done at `web5` layer, but a protocol scoped grant needs to have a "protocol tag" associated with the record also.

Reply:
lets mark this resolved and that can come in integration later

Reply:
I'm supportive of fixing this in a further integration PR.

If you could ADD this issue link to the TODO: https://github.com/TBD54566975/web5-js/issues/827
There will be a PermissionApi class that will help with the various grant operations.

Reply:
fixed